### DISCUSSION TITLE
Change the transform of an entity inside of a filament instance

### DISCUSSION THREAD

**Discussion by davidwarford** [google/filament#4127] [Q&A] [answered]
Hello, I have a instanced filament made out of multiple entities, is it possible to change the transform of one of the entities?

**Comment by romainguy** [comment, marked as answer]
Each `Entity` has its own transform, so yes. But you cannot set a transform on the *primitives* of a renderable entity. Anyway, you can use `TransformManager` to set transforms on entities.

**Reply by davidwarford** [reply]
that's kind of what I want though, I have multiple parts on this model I would like to have move, is their another way to do it

**Reply by romainguy** [reply]
You'd have to split your model into multiple entities/renderables.

**Reply by davidwarford** [reply]
so each filament instance can only have one transform then and I can't change the transform of the individual entities? can I maybe use animation bones?

**Reply by romainguy** [reply]
Each entity can have its own transform, and you can create hierarchies of entities/transforms. What you cannot do is set a transform on a primitive inside an entity. You can also use skinning and morphing yes.
